binder/src/androidTest/java/io/grpc/binder/internal/BinderTransportTest.java

```diff
@@ -48,8 +50,8 @@
 public final class BinderTransportTest extends AbstractTransportTest {
 
   private final Context appContext = ApplicationProvider.getApplicationContext();
-  private final ScheduledExecutorService scheduledExecutorService =
-      Executors.newScheduledThreadPool(2);
+  private final ObjectPool<ScheduledExecutorService> executorServicePool =
+      new FixedObjectPool<>(Executors.newScheduledThreadPool(2));
```

Comment:
This is actually (already) creating a lot of threads. I don't know if they are GCed on Android either (in Java 8 I think they are, but in 9+ they aren't in openjdk). You probably want to make this `static` and/or shut it down in `tearDown()`. That can be a follow-up.

Another option is to use `SharedResourcePool.forResource(GrpcUtil.TIMER_SERVICE)`.

Reply:
Done.